CONTRIBUTING.md

```diff
@@ -345,6 +345,25 @@ with clear instructions on how to install the required libraries.
 Furthermore, if your package requires CGO, it MUST be able to compile and operate in a no op mode
 or report a warning back to the collector with a clear error saying CGO is required to work.
 
+### Breaking changes
+
+Whenever possible, we adhere to semver as our minimum standards. Even before v1, we strive to not break compatibility
+without a good reason. Hence, when a change is known to cause a breaking change, it should be clearly marked in the
+changelog, possibly with a line instructing users how to move forward.
+
+We also strive to perform breaking changes in two stages: 
+
+- when we need to remove something, we mark a feature as deprecated in one version, and remove it in the next one
```

Comment:
Like the point below ("we create the new name first"), I think we should introduce the replacement code (if it is being replaced) before we deprecate existing functionality?

Reply:
I very much agree with this, though it is attached to a line about removing indentifiers.  If we're purely removing something without replacement (and with no intent to ever offer a replacement) then there is nothing to have in place before the deprecation notice.

If, however, we're removing something to be replaced with something else, even if it is not a renaming but a change of design, we should definitely have the replacement in place before removing the existing code.  We should also, where possible, change the existing implementation to forward to the new implementation, or re-implement in terms of the new model, before removal.

Reply:
I see where it was ambiguous and have addressed that.

Reply:
I'm still a bit unclear... Which category (removal vs renaming) would general refactoring fall into? For example, if I split `ListAndWatch()` into `List()` and `Watch()`, would that be covered by the removal or renaming category here?  Does "renaming" cover all refactoring for which there is a similar replacement, or are refactors considered removals + new additions?  What would this imply for https://github.com/open-telemetry/opentelemetry-collector/pull/4672#issuecomment-1017804080?

Reply:
The rules are the same: MUST mark as deprecated in one version, MAY remove in the next. The difference is that renaming causes the deprecated code to call the new one. So, `ListAndWatch` gets deprecated and calls `List()` and `Watch()`, until it eventually gets removed.

Reply:
Can this be marked as resolved, or is there still something I need to clarify in the doc?

Reply:
Can you expand renaming to be clear it covers refactoring as well?
`when renaming or refactoring types, functions, or attributes`...

Reply:
Got it! Done.